src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -298,8 +298,9 @@ internal bool IsEqualToOrDerivedFrom(TypeSymbol type, bool ignoreDynamic, ref Ha
         /// <param name="t2">The other type.</param>
         /// <param name="ignoreCustomModifiersAndArraySizesAndLowerBounds">True to compare without regard to custom modifiers, false by default.</param>
         /// <param name="ignoreDynamic">True to ignore the distinction between object and dynamic, false by default.</param>
+        /// <param name="ignoreTupleNames">True to ignore the distinction between tuple element names, true by default.</param>
         /// <returns>True if the types are equivalent.</returns>
-        internal virtual bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArraySizesAndLowerBounds = false, bool ignoreDynamic = false)
+        internal virtual bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArraySizesAndLowerBounds = false, bool ignoreDynamic = false, bool ignoreTupleNames = false)
```

Comment:
I think the method has evolved enough to wrap the switches into a single `CompareKind` where CompareKind is 
```C#
<Flags>
enum TypeCompareKind
{
       ConsiderEverything: 0,
       IgnoreCustomModifiersAndStuff: 1,
       IgnoreDynamic: 2,
       IgnoreTupleNames 4,
}   
```